backend/controller/observability/observability.go

```diff
@@ -0,0 +1,195 @@
+package observability
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type metricAttributeBuilders struct {
+	moduleName      func(name string) attribute.KeyValue
+	featureName     func(name string) attribute.KeyValue
+	destinationVerb func(name string) attribute.KeyValue
+}
+
+type callMetrics struct {
+	meter    metric.Meter
+	requests metric.Int64Counter
+	failures metric.Int64Counter
+	active   metric.Int64UpDownCounter
+	latency  metric.Int64Histogram
+}
+
+type fsmMetrics struct {
+	meter       metric.Meter
+	active      metric.Int64UpDownCounter
+	transitions metric.Int64Counter
+}
+
+type observableMetrics struct {
+	attributes metricAttributeBuilders
+	calls      *callMetrics
+	fsm        *fsmMetrics
+}
+
+var (
+	metrics = observableMetrics{
+		// TODO: move to a initialization method
+		attributes: metricAttributeBuilders{
+			moduleName: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.module.name", name)
+			},
+			featureName: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.feature.name", name)
+			},
+			destinationVerb: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.verb.dest", name)
+			},
+		},
```

Comment:
It feels like a lot of this could be simplified/shortened to just `const` declarations of the attr names. Since below, it's already a full line, we would save a bit of complexity here and not lose anything below. Ex: 

```
moduleAttr := metrics.attributes.moduleName(fsm.Module)
```
to
```
moduleAttr := attribute.String(ATTR_MODULE, fsm.Module)
```

It's nice having the attributes associated with their metrics, but it's already associated below since we need to actually instantiate all the attrs below anyway. So we don't need to duplicate those associations up here

Reply:
There is a difference between the two options worth noting. Attributes have a type associated with them and the second option decouples name and type. Attributes do have a schema to them and the proposal opens up the possibility of accidental type mismatching - I'm not sure where the break would end up but since there is no schema registration up front any breakage would occur at run-time or collection time. The first option is meant to defend against that error class.

Reply:
Great point. Fortunately, it looks like the typecheck is built into the attr package :) https://pkg.go.dev/go.opentelemetry.io/otel/attribute#String

Reply:
I'm pretty sure that should fail to build but let me confirm

Reply:
yeah confirmed:
```
backend/controller/observability/call.go:36:44: cannot use "msg" (untyped string constant) as int value in argument to attribute.Int
```

Reply:
Got it. Helper functions do make sense for that. Let's just make sure we're only creating those helper functions once for each attribute. My main concern here was with trying to capture the associations of metrics with attributes in the `observableMetrics` struct.